adafruit_wiznet5k/adafruit_wiznet5k_socket.py

```diff
@@ -160,6 +185,34 @@ def listen(self, backlog=None):
         _the_interface.socket_listen(self.socknum, self._listen_port)
         self._buffer = b""
 
+    def accept(self):
+        """Mimic python socket accept for compatibility. The socket where the
+        connection originated is returned while a new socket is allocated and begins
+        listening.
+        """
+        stamp = time.monotonic()
+        while self.status not in (
+            adafruit_wiznet5k.SNSR_SOCK_SYNRECV,
+            adafruit_wiznet5k.SNSR_SOCK_ESTABLISHED,
+        ):
+            if self._timeout > 0 and time.monotonic() - stamp > self._timeout:
+                return None
+            if self.status == adafruit_wiznet5k.SNSR_SOCK_CLOSED:
+                self.close()
+                self.listen()
+
+        new_listen_socknum, addr = _the_interface.socket_accept(self.socknum)
+        current_socknum = self.socknum
+        # Create a new socket object and swap socket nums so we can continue listening
+        client_sock = socket()
+        client_sock.socknum = current_socknum
+        self.socknum = new_listen_socknum
+        self.bind((None, self._listen_port))
+        self.listen()
+        while self.status != adafruit_wiznet5k.SNSR_SOCK_LISTEN:
+            print("Waiting for socket to listen")
```

Comment:
Please do not print within a library. Add a retry loop which `raise`'s or returns if it fails.